File: SGEI-API/SGEI/Models/Master/PersonsxStudent.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace SGEI.Models.Master
{
  public class PersonsxStudent
  {
    [Key]
    public long id { get; set; }

    public long idpersona { get; set; }

    public long idestudiante { get; set; }

    public bool activo { get; set; }

    [ForeignKey(nameof(idpersona))]
    public Person persona { get; set; }

    [JsonIgnore]
    [ForeignKey(nameof(idestudiante))]
    public Student estudiante { get; set; }

    public bool esrepresentante { get; set; }
  }
}

File: src/app/models/master/personsxstudent.ts
import { Person } from "./person";
import { Student } from "./student";

export class PersonxStudent{
    id: number = -1;
    idpersona: number = -1;
    idestudiante: number = -1;
    activo: boolean = false;
    persona: Person = new Person();
    estudiante: Student = new Student();
    esrepresentante: boolean = false;
}